lib/reconcile/vernemq.go

```diff
@@ -315,11 +342,67 @@ func getVerneMQPodSpec(statefulSetName, dataVolumeName string, cr *apiv1alpha1.A
 				Env:            getVerneMQEnvVars(statefulSetName, cr),
 			},
 		},
+		Volumes: getVerneMQVolume(AstarteVerneMQCertificateName, cr),
 	}
 
 	return ps
 }
 
+func getVerneMQVolume(volumeName string, cr *apiv1alpha1.Astarte) []v1.Volume {
+	theVolume := []v1.Volume{}
+
+	// if SSL termination must be handled at VerneMQ level, create the volume to store the certificates
+	if pointy.BoolValue(cr.Spec.VerneMQ.TerminateSSL, false) {
+		theVolume = append(theVolume, v1.Volume{
+			Name: volumeName,
+			VolumeSource: v1.VolumeSource{
+				Secret: &v1.SecretVolumeSource{
+					DefaultMode: pointy.Int32(420),
+					SecretName:  AstarteVerneMQCertificateName,
+					Items: []v1.KeyToPath{
```

Comment:
Do you really need this? In theory, TLS secrets should just work without specifying items explicitly, in the very same way - https://kubernetes.io/docs/concepts/configuration/secret/#tls-secrets

Reply:
It is needed. If omitted, we would end up mounting the files `tls.crt` and `tls.key`, while we need them mounted as `cert` and `privkey` respectively.